packages/dotcom-ui-header/src/__stories__/story.tsx

```diff
@@ -16,104 +14,244 @@ import profileStoryData from './story-data/profile'
 import '../../styles.scss'
 import './demos.scss'
 
-const toggleUserStateOptions = () => boolean('Enable user nav actions', true)
-const toggleVariantOptions = () => radios('Choose variant', { default: 'simple', large: 'large' }, 'simple')
-const toggleShowLogoLink = () => boolean('Render with the logo having a link', false)
-const toggleLoggedIn = () => boolean('User is logged in', false)
-const toggleShowSubNav = () => boolean('Show the sub-navigation component', true)
-const toggleShowStickyHeader = () => boolean('Show the sticky header', true)
-const toggleShowMegaNav = () => boolean('Show the drop-down "mega nav"', true)
-const toggleMobileNav = () => radios('Show mobile nav', { show: '/', hide: '/404' }, '/')
-
 const onReadyCallback = () => {
   // Passing a cors-anywhere hostname to n-topic-search
   // An 'origin' request header will be set on the subsequent fetch request to next-search-api
   // This satisfies the api's cors rules allowing a response to be sent and rendered on localhost
   header.init({ hostName: 'cors-anywhere.herokuapp.com/www.ft.com' })
 }
 
-storiesOf('FT / Header', module)
-  .addDecorator(withKnobs)
-  .add('Default header with drawer', () => {
-    const knobs = {
-      showSubNavigation: toggleShowSubNav(),
-      showMegaNav: toggleShowMegaNav(),
-      showUserNavigation: toggleUserStateOptions(),
-      userIsLoggedIn: toggleLoggedIn(),
-      currentPath: toggleMobileNav(),
-      showLogoLink: toggleShowLogoLink()
-    }
-    storyData.data = { ...storyData.data, currentPath: toggleMobileNav() }
+export default {
+  title: 'FT / Header'
+}
+
+export const DefaultHeaderWithDrawer = (args) => (
+  <OnReady callback={onReadyCallback}>
+    <HeaderSimple {...storyData} {...args} />
+    <Drawer {...storyData} {...args} />
+  </OnReady>
+)
+
+DefaultHeaderWithDrawer.story = {
+  name: 'Default header with drawer'
+}
+DefaultHeaderWithDrawer.args = {
+  showSubNavigation: true,
+  showMegaNav: true,
+  showUserNavigation: true,
+  userIsLoggedIn: false,
+  currentPath: '/',
+  showLogoLink: false
+}
+DefaultHeaderWithDrawer.argTypes = {
+  showSubNavigation: {
+    control: {
+      type: 'boolean'
```

Comment:
my understanding was that for simple controls like boolean values, it would be able to infer the control type, so we wouldn't need an `argType` for these, only the ones where the inferred type isn't helpful (like `currentPath` probably)? think this would remove a lot of the duplication.

Reply:
ohhh, yeah maybe this and the disabling could do the trick, thank you.

Reply:
Updated...